src/app/maptools/qgsmaptoolshapecircularstringabstract.cpp

```diff
@@ -191,7 +191,21 @@ void QgsMapToolShapeCircularStringAbstract::addCurveToParentTool()
 {
   QgsCircularString *c = new QgsCircularString();
   c->setPoints( mPoints );
-  mParentTool->addCurve( c );
+
+  // Check whether to draw the circle as a polygon or a circular string
+  const QgsCoordinateReferenceSystem layerCrs = mParentTool->layer()->crs();
```

Comment:
Note for myself -- double check if layer() can be nullptr here before merging

Reply:
@YoannQDQ this is indeed the case -- layer() may be nullptr in some circumstances. Eg QgsMapToolProfileCurve::layer(). In this case the map canvas crs should be used.

Reply:
Ok